lib/generator.js

```diff
@@ -1281,8 +1281,10 @@
                                     (targetWidth ? (targetWidth / paddedInputWidth) : targetScaleY)),
 
             // Effects are not scaled when the transformation is non-uniform
-            effectsScaled       =
-                (paddedOutputWidthFloat / paddedInputWidth) === (paddedOutputHeightFloat / paddedInputHeight),
+            paddedFloatRatioDiff = Math.abs(paddedOutputWidthFloat / paddedInputWidth -
+                                            paddedOutputHeightFloat / paddedInputHeight),
+            effectsScaled       = (targetWidth || targetHeight) ? paddedFloatRatioDiff < Number.EPSILON :
```

Comment:
Maybe I'm missing something, but as near as I can tell, `Number.EPSILON` is undefined.

Right before this var decl, I added the logging statement `_logger.debug("Number.EPSILON is", Number.EPSILON);` and it printed:

`[debug:core 20:17:45.877 generator.js:1256:17] Number.EPSILON is undefined`

Also, `EPSILON` is not a built in property of Number, and I don't think we should be hanging new properties off of built-ins.